### ISSUE TITLE
Rerun failures and logic onTestFailure doesn't work for the tests with dependsOnMethods

### ISSUE THREAD

**Issue by kolyakrasnik** [qaprosoft/carina#731]
**Describe the bug**
In case if we have the next structure:

```
@Test()
public void **test1**()
{
actions();
}

@Test(dependsOnMethods = "**test1**")
public void test2()
{
actions();
}
```
retry doesn't work properly.
As far as method **test1** has dependencies we don't quit drivers (logic from CarinaListener.class)
```
if (automaticDriversCleanup && !hasDependencies(result)) {
                quitDrivers(Phase.BEFORE_METHOD, Phase.METHOD);
            }
```
So, we are trying perform rerun failure with already started driver.

Carina version 6.1.20.

**Comment by kolyakrasnik** [comment]
PR: https://github.com/qaprosoft/carina/pull/733 
Was added check if result == 2 (failure) then quit driver. So, we'll close driver anyway for failure.

**Comment by vdelendik** [comment]
fix should be in 6.1.21+
testing is available via 6.1.21-SNAPSHOT core build
